src/PhotometryFit.cc

```diff
@@ -216,78 +216,86 @@ void PhotometryFit::offsetParams(Eigen::VectorXd const &delta) {
+            double residual = flux - fittedStar->getFlux();
             double chi2Val = std::pow(residual / sigma, 2);
-            ofile << measuredStar.x << "\t" << measuredStar.y << "\t" << fs->getMag() << "\t"
-                  << measuredStar.getInstFlux() << "\t" << measuredStar.getInstFluxErr() << "\t"
-                  << measuredStar.getFlux() << "\t" << measuredStar.getFluxErr() << "\t" << flux << "\t"
-                  << fluxErr << "\t" << fs->getFlux() << "\t" << jd << "\t" << fs->color << "\t"
-                  << fs->getIndexInMatrix() << "\t" << fs->x << "\t" << fs->y << "\t" << chi2Val << "\t"
-                  << fs->getMeasurementCount() << "\t" << ccdImage.getCcdId() << "\t" << ccdImage.getVisit()
-                  << std::endl;
+
+            ofile << measuredStar->x << separator << measuredStar->y << separator << fittedStar->getMag()
+                  << separator << measuredStar->getInstFlux() << separator << measuredStar->getInstFluxErr()
+                  << separator << measuredStar->getFlux() << separator << measuredStar->getFluxErr()
+                  << separator << flux << separator << fluxErr << separator << fittedStar->getFlux()
+                  << separator << jd << separator << fittedStar->color << separator
+                  << fittedStar->getIndexInMatrix() << separator << fittedStar->x << separator
+                  << fittedStar->y << separator << chi2Val << separator << fittedStar->getMeasurementCount()
+                  << separator << ccdImage->getCcdId() << separator << ccdImage->getVisit() << std::endl;
         }  // loop on measurements in image
     }      // loop on images
 }
+
+void PhotometryFit::makeRefResultTuple(std::string const &tupleName) const {
+    std::ofstream ofile(tupleName.c_str());
+    std::string separator = "\t";
+    ofile << "#ra" << separator << "dec " << separator << "mag" << separator << "color" << separator
+          << "refFlux" << separator << "refFluxErr" << separator << "fittedFlux" << separator
+          << "fittedFluxErr" << separator << "fsindex" << separator << "chi2" << separator << "nm"
+          << std::endl;
+    ofile << "#coordinates of fittedStar" << separator << separator << "magnitude" << separator
+          << "currently unused" << separator << "default refStar flux" << separator
+          << "default refStar fluxErr" << separator << "fittedStar flux" << separator << "fittedStar fluxErr"
+          << separator << "unique index of the fittedStar" << separator
+          << "refStar contribution to Chi2 (2D dofs)" << separator
+          << "number of measurements of this FittedStar" << std::endl;
+    // The following loop is heavily inspired from AstrometryFit::computeChi2()
+    const FittedStarList &fittedStarList = _associations->fittedStarList;
+    for (auto const &fittedStar : fittedStarList) {
+        const RefStar *refStar = fittedStar->getRefStar();
+        if (refStar == nullptr) continue;
```

Comment:
Out of curiosity, is it rare or frequent for the refStar == nullptr? What condition?

Reply:
Any time there isn't a refStar for that fittedStar (which could be most of them).